File: 0290-word-pattern/0290-word-pattern.js
/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
var wordPattern = function(pattern, s) {
    list = s.split(" ");
    console.log(list);

    var dic1 = {}
    var dic2 = {}
    if(list.length != pattern.length)return false;
    for(var i = 0 ; i < list.length; i++){
        if(list[i] in dic1){}
        else dic1[list[i]]=i;
        if(pattern[i] in dic2){}
        else dic2[pattern[i]]=i;
        if(dic1[list[i]] != dic2[pattern[i]])return false; 
    }
    return true;
};

File: 0290-word-pattern/0290-word-pattern.cpp
class Solution {
public:
    bool wordPattern(string pattern, string s) {
        vector<string>v;
        stringstream ss(s);
        while(ss>>s)v.push_back(s);
        if(v.size() != pattern.size())return 0;
        unordered_map<string,int>mp2;
        unordered_map<char,int>mp;

        for(int i = 0 ; i < v.size() ; i++){
            if(mp.find(pattern[i])==mp.end())mp[pattern[i]]=i;
            if(mp2.find(v[i])==mp2.end())mp2[v[i]]=i;
            if(mp2[v[i]] != mp[pattern[i]])return 0;
        }
        return 1;
    }
};